src/counting_sort.rs

```diff
@@ -0,0 +1,95 @@
+// Implements http://rosettacode.org/wiki/Sorting_algorithms/Counting_sort
+#![allow(unstable)]
+
+fn counting_sort(array: &mut [i32]) {
+
+	// nothing to do for arrays shorter than 2
+	if array.len() < 2 {
+		return
+	}
+
+	// first we find the minimum and maximum value
+	let mut min = array[0];
+	let mut max = array[0];
+	for e in array.iter() {
+		if *e < min {
+			min = *e;
+		}
+		if *e > max {
+			max = *e;
+		}
+	}
+
+	// then we count occurences of values
+	let size = (max - min + 1) as usize;
+	let mut count: Vec<i32> = Vec::with_capacity(size);
+	count.resize(size, 0);
```

Comment:
vec::from_fn is removed ? 
otherwise this is shorter: Vec::from_fn(size, |_| 0);